### PR TITLE
Add basic support for ayaneo slide.

### PR DESCRIPTION
**Pull request by DevL0rd**
Resolves issues #997, and seemingly #976

All buttons working, glyphs are correct, gyro is corrected.

Issues:

I'm not sure how to get the correct ECDetails, as such I have disabled fan control.

However I am interested in learning what I can do to get these functions working :)

## Summary by CodeRabbit

- **New Features**
    - Introduced support for the AYANEO SLIDE device, including device-specific settings, capabilities, and custom key mappings.
    - Added support for a new device "AS01" by introducing a corresponding class.
    - Added a new image resource file named `device_aya_slide.png` to the project.

### CODE REVIEW

File: HandheldCompanion/Devices/AYANEO/AYANEOSLIDE.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        JoystickLeft = 1,
+        JoystickRight = 2,
+        JoystickBoth = 3,
+        Keyboard = 4,
+    }
+
+    private static byte[] AYA_ZONES = new byte[] { 4 };
+    private static byte[] STICK_ZONES = new byte[] { 1, 2, 3, 4 };
+
+    private Color color = Color.FromRgb(0, 0, 0);
```
[the remark this answers is not in the document]
Comment (DevL0rd) [reply HandheldCompanion/Devices/AYANEO/AYANEOSLIDE.cs]: fixed

File: HandheldCompanion/Devices/AYANEO/AYANEOSLIDE.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    private enum LEDGroup
+    {
+        JoystickLeft = 1,
+        JoystickRight = 2,
+        JoystickBoth = 3,
+        Keyboard = 4,
+    }
+
+    private static byte[] AYA_ZONES = new byte[] { 4 };
+    private static byte[] STICK_ZONES = new byte[] { 1, 2, 3, 4 };
```
[the remark this answers is not in the document]
Comment (DevL0rd) [reply HandheldCompanion/Devices/AYANEO/AYANEOSLIDE.cs]: fixed

**Comment by DevL0rd** [comment]
Ok It's ready :D Thanks for showing me around the repo guys <3

**Comment by CasperH2O** [comment]
For an image, what I do is get a nice product image online, then in Paint.net, make the area around it transparant. Then realize that the width is the same as the generic device picture and save as png.

For the fan, we use RW everything and check which adresses Ayaspace adjusts and which change based on fan speed.

https://github.com/Valkirie/HandheldCompanion/wiki/Tools#rweverything

Did you get the LEDs working?

**Comment by DevL0rd** [comment]
@CasperH2O using HW everything may be a bit outta my league hahaha

could you give a brief breakdown of some steps to follow to get the addresses?

I'll try the same in the end with the lighting controls. maybe in a different PR

**Comment by DevL0rd** [comment]
ah I get it now. @CasperH2O 

I see the button presses from me hitting the extra buttons. but I really can't find a correlation between me settings fans to 100 and to auto.

any suggestions?

**Comment by Valkirie** [comment]
@DevL0rd may I ask you to share some photos of HC running on AYANEO Slide. I'd like to use them to advertise forthcoming compatibility :)

**Comment by DevL0rd** [comment]
yes sir @Valkirie , I can do so when I am home, right after I make a nice transparent image for the device preview in the about menu. 🙂👍

**Comment by DevL0rd** [comment]
Screenshot_2024-02-17_191155
Screenshot_2024-02-17_191226
@Valkirie Ready!

**Comment by Valkirie** [comment]
Thanks! @CasperH2O shall we merge as-is or do you want to wait until you get yours ?

**Comment by CasperH2O** [comment]
Nice pictures! 

Im good with doing this set now and the fan and LED later when I have a device.

Can we squash this PR in one commit though, seems cleaner?

**Comment by DevL0rd** [comment]
idk how to squash haha.

but yeah I figure do fan and led after I'm going to play with it myself and see if I learn something meanwhile.

at least for now everything else works.

Just a note from that issue, ayaneo ships with a broken driver for the gyro and accelerometer. have to install the latest driver or gyro just doesn't function at all.

**Comment by CasperH2O** [comment]
Depending on your Git client, squash and merge commit instructions for GitHub Desktop Client: https://docs.github.com/en/desktop/managing-commits/squashing-commits-in-github-desktop

An alternative quick and easy way is to copy the changed files from your repo, make a new branch from main, paste the files, test they work, commit and PR.

**Comment by CasperH2O** [comment]
Correction, using GitHubs squash and merge still creates a merge commit, which we dont want.

**Comment by DevL0rd** [comment]
alright guess I'm using GitHub desktop now haha.

all done!

**Comment by CasperH2O** [comment]
Excellent!

**Pull request merged**